string_filters.hpp

```diff
@@ -136,6 +150,7 @@ struct canonicalize_t {
         std::ranges::copy(parts_view, std::back_inserter(processed_words));
         // Join the processed words with an underscore
         return strings::join(processed_words, "_");
+#endif
```

Comment:
```suggestion
#endif
        return strings::join(processed_words, "_");
```

Doing it this way just changes the way processed_words are built and keeps the rest of the logic uniform, which is clearer intent imho.